main.go

```diff
@@ -59,18 +61,26 @@ func (cw CSVWorker) subscribe(ctx context.Context, client *pubsub.Client) {
 			logger.Info("Message delivery attempted", zap.Int("delivery attempts", *msg.DeliveryAttempt))
 		}
 
-		sample := msg.Data
+		data := msg.Data
 		attribute := msg.Attributes
 		sampleSummaryId, ok := attribute["sample_summary_id"]
 		if ok {
 			logger.Info("about to process sample", zap.String("sampleSummaryId", sampleSummaryId))
-			err := processSample(sample, sampleSummaryId, msg)
+			line := readSampleLine(data)
+			sampleUnitId, err := processSample(line, sampleSummaryId, msg)
 			if err != nil {
 				logger.Error("error processing sample - nacking message", zap.Error(err))
 				//after x number of nacks message will be DLQ
 				msg.Nack()
 			} else {
-				logger.Info("sample processed - acking message")
+				//now the sample has been create, lets create the associated party
+				err := processParty(line, sampleSummaryId, sampleUnitId, msg)
+				if err != nil {
+					logger.Error("error processing party - nacking message", zap.Error(err))
+					//after x number of nacks message will be DLQ
+					msg.Nack()
```

Comment:
Less a comment and more a question.  What happens when we nack this?  We've made something in sample but not in party.  Does that mean it's now orphaned?  (I think it is), is this the same as it was before? (also probably yes?) and do we recover from it by just trying it again?

Reply:
That's a good spot, in this scenario it would retry and sample would return a 409. It would then go on to create a party but the sample unit id would be blank which is incorrect.

I look at how to fix this, might need to call the sample service to get the ID of the previously created one.